src/test/java/com/exasol/adapter/dialects/bigquery/BigQuerySqlDialectTest.java

```diff
@@ -122,9 +129,23 @@ void testGetSupportedProperties() {
                         DEBUG_ADDRESS_PROPERTY, LOG_LEVEL_PROPERTY, BIGQUERY_ENABLE_IMPORT_PROPERTY));
     }
 
-    @Test
-    void testApplyQuote() {
-        assertThat(this.dialect.applyQuote("tableName"), equalTo("`tableName`"));
+    @CsvSource({ "5Customers, `5Customers`", //
+            "_dataField1, `_dataField1`", //
+            "tableName, `tableName`", //
+            "\" table123 `, `\" table123 \\``" //
+    })
+    @ParameterizedTest
+    void testApplyQuote(final String unquoted, final String quoted) {
+        assertThat(this.dialect.applyQuote(unquoted), equalTo(quoted));
+    }
+
+    @ValueSource(strings = { "ab:'ab'", "a'b:'a\\'b'", "a''b:'a\\'\\'b'", "'ab':'\\'ab\\''" })
```

Comment:
Please add the same test cases I suggested for the other dialect with \ escaping.